spacepy/time.py

```diff
@@ -2277,6 +2281,17 @@ def _read_leaps():
         mon[i] = int(np.where(months == line.split()[1])[0][0] + 1)
         day[i] = int(line.split()[2])
 
+    # Check for out of date. The leap second bulletin comes every
+    # six months, and that contains information through the following
+    # leap second (end of June/Dec)
+    lastknown = max(
+        mtime, datetime.datetime(int(year[-1]), int(mon[-1]), int(day[-1])))
+    goodthrough = datetime.datetime(lastknown.year + int(lastknown.month > 6),
+                                    1 if lastknown.month > 6 else 7, 1)
```

Comment:
It may pay to just be less smart here. `goodthrough = lastknown + datetime.timedelta(months=7)` is likely good enough to catch really out of date systems.